### DISCUSSION TITLE
Build breaking on vercel due to @sveltejs/adapter-vercel peer dependency issue

### DISCUSSION THREAD

**Discussion by gorayaa66** [sveltejs/kit#11965] [Q&A] [answered]
### Describe the bug

After migrating to sveltekit 2, it was getting deployed successfully but suddenly now i'm getting peer dependency issue for @sveltejs/adapter-vercel.
 
 ### Here is the list of packages i'm using.
"@sveltejs/adapter-vercel": "^4.0.0",
"@sveltejs/kit": "^2.0.0",
"svelte": "^4.0.0",

### Reproduction

I'm going to paste all packages here, in case of reproduction of this bug.

                "@sveltejs/adapter-vercel": "^4.0.0",
		"@sveltejs/kit": "^2.0.0",
		"@sveltejs/vite-plugin-svelte": "^3.0.0",
		"@tailwindcss/forms": "^0.5.3",
		"@tailwindcss/typography": "^0.5.9",
		"@testing-library/jest-dom": "^6.4.2",
		"@testing-library/svelte": "^4.1.0",
		"@types/dompurify": "^3.0.1",
		"@types/jest": "^29.5.12",
		"@types/jsonwebtoken": "^9.0.2",
		"@types/mailgun-js": "^0.22.13",
		"@types/marked": "^4.0.8",
		"@typescript-eslint/eslint-plugin": "^5.45.0",
		"@typescript-eslint/parser": "^5.45.0",
		"autoprefixer": "^10.4.13",
		"eslint": "^8.28.0",
		"eslint-config-prettier": "^8.5.0",
		"eslint-plugin-svelte": "^2.35.1",
		"jsdom": "^24.0.0",
		"postcss": "^8.4.21",
		"prettier": "^2.8.4",
		"prettier-plugin-svelte": "^2.10.1",
		"prettier-plugin-tailwindcss": "^0.2.3",
		"stripe": "^11.17.0",
		"supabase": "^1.61.3",
		"svelte": "^4.0.0",
		"svelte-check": "^3.4.3",
		"svelte-stripe": "^0.0.22",
		"tailwindcss": "^3.4.1",
		"tslib": "^2.4.1",
		"typescript": "^5.0.0",
		"vite": "^5.0.0",
		"vitest": "^1.2.2"

### Logs

```Shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @supabase/auth-helpers-sveltekit@0.9.4
npm ERR! Found: @sveltejs/kit@2.5.2
npm ERR! node_modules/@sveltejs/kit
npm ERR!   dev @sveltejs/kit@"^2.0.0" from the root project
npm ERR!   peer @sveltejs/kit@"^2.0.0" from @sveltejs/adapter-vercel@4.0.5
npm ERR!   node_modules/@sveltejs/adapter-vercel
npm ERR!     dev @sveltejs/adapter-vercel@"^4.0.5" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @sveltejs/kit@"^1.15.2" from @supabase/auth-helpers-sveltekit@0.9.4
npm ERR! node_modules/@supabase/auth-helpers-sveltekit
npm ERR!   @supabase/auth-helpers-sveltekit@"^0.9.4" from the root project
```

### System Info

```Shell
I'm encountering an build failure during Vercel deployment, even though my application runs perfectly on localhost development. The "error log" pasted above is available for reference.
```

### Severity

blocking an upgrade

### Additional Information

Can you help me identify and resolve the issue?

**Comment by teemingc** [comment, marked as answer]
Hi, I've transferred your issue to discussions as this is a more appropriate place for it.

The problem is explained in the error logs:
```
npm ERR! Could not resolve dependency:
npm ERR! peer @sveltejs/kit@"^1.15.2" from @supabase/auth-helpers-sveltekit@0.9.4
npm ERR! node_modules/@supabase/auth-helpers-sveltekit
npm ERR!   @supabase/auth-helpers-sveltekit@"^0.9.4" from the root project
```
This says that the `@supabase/auth-helpers-sveltekit@0.9.4` supabase dependency requires `@sveltejs/kit@"^1.15.2"`, that is SvelteKit 1, but your project has installed SvelteKit 2.

Fortunately, there is a newer version of the superbase dependency that supports SvelteKit 2. You'll need to update to a later version https://www.npmjs.com/package/@supabase/auth-helpers-sveltekit?activeTab=readme (latest version is 0.12.0).

You can read about the supabase package's peer dependencies (other packages and versions it requires) in it's `package.json` file. https://github.com/supabase/auth-helpers/blob/main/packages/sveltekit/package.json just scroll down to the `peerDependencies` part at the bottom of the file.
